compiler/rustc_interface/src/passes.rs

```diff
@@ -795,6 +793,8 @@ pub fn create_global_ctxt<'tcx>(
                 untracked,
                 dep_graph,
                 queries.on_disk_cache.as_ref().map(OnDiskCache::as_dyn),
+                local_providers,
+                extern_providers,
```

Comment:
these could be exposed from the `QueryEngine`, or if that is a performance issue, we could make the current `QueryEngine` uses be a struct with a trailing `dyn QueryEngine` field to have quick access to the fields.